collector/lib/HostHeuristics.cpp

```diff
@@ -185,13 +186,74 @@ class GardenLinuxHeuristic : public Heuristic {
   }
 };
 
+class EBPFPermissionsHeuristic : public Heuristic {
+  // On some platforms, SELinux is known to cause some problems with map creation.
+  // This heuristic will check whether map creation is possible, and if not it
+  // will switch to Kernel Module based collection, logging this as a warning.
+  //
+  // While this issue is known to affect RHEL-7, the heuristic is platform
+  // agnostic to ensure coverage of all platforms.
+  void Process(HostInfo& host, const CollectorConfig& config, HostConfig* hconfig) const {
+    if (!config.UseEbpf() || !host.HasEBPFSupport()) {
+      return;
+    }
+
+    CLOG(DEBUG) << "Attempting to create eBPF map...";
+
+    if (canCreateMap()) {
+      CLOG(DEBUG) << "eBPF map creation successful.";
+      return;
+    }
+
+    CLOG(WARNING) << "Unable to create eBPF map, switching to KernelModule collection";
+    CLOG(WARNING) << "To enable eBPF, check your SELinux settings.";
+    hconfig->SetCollectionMethod("kernel-module");
```

Comment:
I could see that we do this in a lot of places, so this is just a general remark: shouldn't we use enumerated type for the collection method, instead of strings ?

Reply:
Yes, I was just thinking loud :)